### PR TITLE
Add automated tests for AlloyDB instance 

### PR DESCRIPTION
**Pull request by GauravJain21**
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
Add automated tests for AlloyDB instance
```

### CODE REVIEW

**Comment by zli82016** [comment]
@GauravJain21 , the new tests failed, and you have to fix them.

**Comment by GauravJain21** [comment]
Hey @zli82016 
Thanks for pointing this out. This is an in-progress PR. I am fixing the tests and will notify once that is done.

**Comment by GauravJain21** [comment]
Hey @zli82016 
All the tests are passing now. Can you please review, approve, and merge the PR?
Best,
Gaurav Jain

File: mmv1/third_party/terraform/tests/resource_alloydb_instance_test.go [hunk 1]
```diff
@@ -90,3 +90,313 @@ resource "google_service_networking_connection" "vpc_connection" {
...
+}
+
+func testAccAlloydbInstance_autoExplainDisabledInPrimaryInstance(context map[string]interface{}) string {
+	return Nprintf(`
+resource "google_alloydb_instance" "primary" {
+  cluster       = google_alloydb_cluster.default.name
+  instance_id   = "tf-test-alloydb-instance%{random_suffix}"
+  instance_type = "PRIMARY"
+  database_flags = {
+	  "alloydb.enable_auto_explain" = "false"
```
Comment (zli82016) [inline mmv1/third_party/terraform/tests/resource_alloydb_instance_test.go:371]: indentation is weird here.

**Review by zli82016 — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**